src/app/charts/line-graph.component.ts

```diff
@@ -182,4 +178,69 @@ export class LineGraphComponent {
     return [slope, intercept, rSquare];
   }
 
+  private drawMinMax(): void {
+    if (self.min || self.max) {
+      // Prepare standard variables
+      var x1 = self.xRange[1];
+      var x2 = self.xRange[0];
+      var y = D3.max(self.yData)/2;
+      var xscale = D3.scaleBand()
+                    .range([0, self.width])
+                    .padding(0)
+                    .domain(self.extractedData.map(function(d) { return d.date; }));
+
+      if (self.min && self.minVal) {
+        // Draw min threshold line
+        var minData = [{"date":x1, "value": self.minVal }, {"date": x2, "value": self.minVal }];
+        self.drawLine(minData, "minline");
+
+        // Prepare data for bar graph
+        var minBars = self.extractedData.map(function(datum) {
+            var val;
+            self.minVal > datum["value"]? val = y*2 : val = 0;
+            return { "date": datum["date"], "value": val }
+        });
+
+        // Add bar graph
+        self.svg.selectAll(".min-bar")
+          .data(minBars)
+          .enter().append("rect")
+          .attr("class", "min-bar")
+          .attr("x", function(d) { return xscale(d.date); })
+          .attr("width", xscale.bandwidth())
+          .attr("y", function(d) { return self.yScale(d.value); })
+          .attr("height", function(d) { return self.height - self.yScale(d.value); });
+      }
+
+      if (self.max && self.maxVal) {
```

Comment:
Looks like you're copying almost the exact same code here for the max operation as above for the min operation. Can you think of a way to abstract this so that you only have to call a function twice rather than duplicate code?